mars/deploy/oscar/session.py

```diff
@@ -1697,6 +1701,27 @@ def __exit__(self, *_):
         self.close()
 
 
+async def _execute_show_progress(
+    execution_info: ExecutionInfo,
+    progress_bar: ProgressBar,
+    progress_update_interval: Union[int, float],
+    cancelled: asyncio.Event,
+):
+    with progress_bar:
+        while not cancelled.is_set():
+            done, pending = await asyncio.wait(
```

Comment:
Better to use `shield` as the older one.

Reply:
The original implementation uses `wait_for`, so the target coroutine should be shielded to avoid cancellation.
From Python doc: https://docs.python.org/3/library/asyncio-task.html#asyncio.wait_for

> If a timeout occurs, it cancels the task and raises [asyncio.TimeoutError](https://docs.python.org/3/library/asyncio-exceptions.html#asyncio.TimeoutError).
> 
> To avoid the task [cancellation](https://docs.python.org/3/library/asyncio-task.html#asyncio.Task.cancel), wrap it in [shield()](https://docs.python.org/3/library/asyncio-task.html#asyncio.shield).

This PR uses `wait`, so it does not need the `shield`.
From Python doc: https://docs.python.org/3/library/asyncio-task.html#asyncio.wait

> Unlike [wait_for()](https://docs.python.org/3/library/asyncio-task.html#asyncio.wait_for), wait() does not cancel the futures when a timeout occurs.

Reply:
Sure, looks good to me now.